Add goodput tutorial (#46)
triton-inference-server/perf_analyzer: genai-perf/docs/goodput.md

 * Redistributions in binary form must reproduce the above copyright
   notice, this list of conditions and the following disclaimer in the
   documentation and/or other materials provided with the distribution.
 * Neither the name of NVIDIA CORPORATION nor the names of its
   contributors may be used to endorse or promote products derived
   from this software without specific prior written permission.
THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->

> # Context

Reviewer: For consistency with our other documents, I would have this title be a verb phrase e.g. "Benchmark Goodput with GenAI-Perf" and then have everything else be one header level down (i.e. "##" and below) since there should only be one title. You can then get rid of of the context title and instead include that background information directly underneath the title.

Author: Updated tutorial according to suggestions above.

Goodput, defined as the number of completed requests per second that meet the
constraints, provides an enhanced measure of AI serving performance by
accounting for both cost efficiency and user satisfaction.

# Tutorials

> This is a tutorial on how to benchmark models using goodput.

Reviewer: This should be plural to match the title.

```suggestion
Below you can find tutorials on how to benchmark different models 
using goodput.
```

## Examples

- [LLM Examples](#LLM)

- [Embedding Model Examples](#embeddings)

## Profile LLM Goodput<a id="LLM"></a>

> ### Run GPT2 on OpenAI Chat Completions API-compatible server

Reviewer: ```suggestion
### Run GPT2 on an OpenAI Chat Completions API-compatible server
```
